src/migration/utilities.js

```diff
@@ -1,23 +1,33 @@
-function initializeContract(path, deployer, accounts) {
-  const Contract = require("truffle-contract")
+function initializeContract(path, deployer, account) {
+  const Contract = require("@truffle/contract")
 
   const contract = Contract(require(path))
   contract.setProvider(deployer.provider)
   contract.setNetwork(deployer.network_id)
   // For some reason the automatic value calculation is not working, hence we do:
   contract.defaults({
-    from: accounts[0],
+    from: account,
     gas: 6.5e6,
   })
   return contract
 }
 
-function getDependency(artifacts, network, deployer, account, path) {
+async function getDeployedDependency(artifacts, deployer, account, path) {
+  let contract
```

Comment:
Can this method reuse `getArtifact`?

Reply:
I was not able to do it, as here, we don't need `await contract.deployed()` in the try and catch